components/content-service/pkg/service/usagereport-service.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package service
+
+import (
+	"context"
+
+	"github.com/opentracing/opentracing-go"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/common-go/tracing"
+	"github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/content-service/api/config"
+	"github.com/gitpod-io/gitpod/content-service/pkg/storage"
+)
+
+const (
+	bucketName = "usage-reports"
+)
+
+// UsageReportService implements UsageReportServiceServer
+type UsageReportService struct {
+	cfg config.StorageConfig
+	s   storage.PresignedAccess
+
+	api.UnimplementedUsageReportServiceServer
+}
+
+// NewUsageReportService create a new usagereport service
+func NewUsageReportService(cfg config.StorageConfig) (res *UsageReportService, err error) {
+	s, err := storage.NewPresignedAccess(&cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &UsageReportService{cfg: cfg, s: s}, nil
+}
+
+// UploadURL provides a URL to which clients can upload the content via HTTP PUT.
+func (us *UsageReportService) UploadURL(ctx context.Context, req *api.UsageReportUploadURLRequest) (resp *api.UsageReportUploadURLResponse, err error) {
```

Comment:
Is it possible to write a unit test for the service? Can we inject mock storage?

Reply:
I've added a test using mocked storage in 145519c53.

In general I'm sceptical of the value of these 'mockist' style tests. All we're doing with the test is is ensuring that we pass the right bucket and filenames through to the dependencies and return the upload URL unmodified from the dependency. 

There isn't any logic being tested there.

Reply:
That's fair. Mocks tend to do that. For me the value is that next time we make a change, we don't have a reason to say we won't add a test because there wasn't one in the first place.